node/full_node_integration_test.go

```diff
@@ -321,36 +318,39 @@ func testSingleAggreatorSingleFullNodeSingleLightNode(t *testing.T) {
 	ds, _ := store.NewDefaultInMemoryKVStore()
 	_ = dalc.Init([8]byte{}, nil, ds, log.TestingLogger())
 	_ = dalc.Start()
-	sequencer, _ := createNode(aggCtx, 0, false, true, false, keys, &wg, t)
-	fullNode, _ := createNode(ctx, 1, false, false, false, keys, &wg, t)
+	sequencer, _ := createNode(aggCtx, 0, false, true, false, keys, t)
+	fullNode, _ := createNode(ctx, 1, false, false, false, keys, t)
 
 	sequencer.(*FullNode).dalc = dalc
 	sequencer.(*FullNode).blockManager.SetDALC(dalc)
 	fullNode.(*FullNode).dalc = dalc
 	fullNode.(*FullNode).blockManager.SetDALC(dalc)
 
-	lightNode, _ := createNode(ctx, 2, false, false, true, keys, &wg, t)
+	lightNode, _ := createNode(ctx, 2, false, false, true, keys, t)
 
 	require.NoError(sequencer.Start())
 	require.NoError(fullNode.Start())
 	require.NoError(lightNode.Start())
 
-	time.Sleep(3 * time.Second)
+	require.NoError(waitForAtLeastNBlocks(sequencer.(*FullNode), 2))
 
 	n1h := sequencer.(*FullNode).hExService.headerStore.Height()
 	aggCancel()
 	require.NoError(sequencer.Stop())
 
-	time.Sleep(3 * time.Second)
-
-	n2h := fullNode.(*FullNode).hExService.headerStore.Height()
-	n3h := lightNode.(*LightNode).hExService.headerStore.Height()
+	require.NoError(testutils.Retry(300, 100*time.Millisecond, func() error {
+		n2h := fullNode.(*FullNode).hExService.headerStore.Height()
+		n3h := lightNode.(*LightNode).hExService.headerStore.Height()
```

Comment:
is there not a common node interface that we can use so that we can have 1 `Height` method?

then the helpers could be re-used for light nodes as well.

Reply:
Yes there is

Could either have the helper try casting it to both types, or add a new method to the interface. The former is probably the way to go, since this method is just for testing.

Reply:
made a helper `getNodeHeight(node Node)`